django/test/testcases.py

```diff
@@ -119,24 +116,20 @@ def __exit__(self, exc_type, exc_value, traceback):
         template_rendered.disconnect(self.on_template_render)
         if exc_type is not None:
             return
-
-        if not self.test():
-            message = self.message()
-            if self.rendered_templates:
-                message += ' Following templates were rendered: %s' % (
-                    ', '.join(self.rendered_template_names)
-                )
-            else:
-                message += ' No template was rendered.'
-            self.test_case.fail(message)
+        self.test_case._assert_template_used(
+            self.template_name, self.rendered_template_names, self.msg_prefix, self.count)
 
 
 class _AssertTemplateNotUsedContext(_AssertTemplateUsedContext):
-    def test(self):
-        return self.template_name not in self.rendered_template_names
 
-    def message(self):
-        return '%s was rendered.' % self.template_name
+    def __exit__(self, exc_type, exc_value, traceback):
+        template_rendered.disconnect(self.on_template_render)
+        if exc_type is not None:
+            return
+        self.test_case.assertFalse(
+            self.template_name in self.rendered_template_names,
+            self.msg_prefix + "Template '%s' was used unexpectedly in rendering the response" % self.template_name
```

Comment:
you could probably use the same pattern here and deduplicate the message.

Reply:
I did consider it, but not sure it is worth the added code complexity of another method in the `SimpleTestCase` just to deduplicate the message (that weren't deduplicated to begin with); however in the `assertTemplateUsed` case there is also a lot of additional logic that I felt worth deduping.